### PR TITLE
Fixed ft_memmove.c compilation errors (with flags -Wall -Wextra -Werror)

### PR DESCRIPTION
**Pull request by Riescent**
Some variables were used without being initialized

### CODE REVIEW

File: templates/libft/ft_memmove.c [hunk 1]
```diff
@@ -70,7 +70,8 @@ void	memmove_int_type(int dest, int src, size_t n, int overlap)
 		return_ptr = ft_memmove(&dest, &src, n);
 	}
 	print_bytes(return_ptr, n);
-	free(int_ptr);
+	if (overlap)
```
Comment (Riescent) [inline templates/libft/ft_memmove.c:73]: `int_ptr` is uninitialized when `if (overlap)` above (line 60) evaluates to false

File: templates/libft/ft_memmove.c [hunk 2]
```diff
@@ -105,10 +107,12 @@ int	main(int argc, char **argv)
 		n = atoi(argv[4]);
 		if (!strcmp(argv[2], "src++"))
 			overlap = 1;
+		else
+			overlap = 0;
```
Comment (Riescent) [inline templates/libft/ft_memmove.c:111]: `overlap` is uninitialized when `if (!strcmp(argv[2], "src++"))` above (line 106/108) evaluates to false

File: templates/libft/ft_memmove.c [hunk 3]
```diff
@@ -92,7 +93,8 @@ void	memmove_float_type(float dest, float src, size_t n, int overlap)
 		return_ptr = ft_memmove(&dest, &src, n);
 	}
 	print_bytes(return_ptr, n);
-	free(float_ptr);
+	if (overlap)
+		free(float_ptr);
```
Comment (Riescent) [inline templates/libft/ft_memmove.c:97]: `float_ptr` is uninitialized when `if (overlap)` above (line 82/83) evaluates to false

**Comment by tdameros** [comment]
Good job, thanks ! I merge it 😀

**Pull request merged**